app/models/middleware_server_performance.rb

```diff
@@ -16,7 +16,11 @@ def self.build_results_for_report_middleware(options)
     start_time, end_time, interval = parse_time_interval(options)
     results = []
     MiddlewareServer.find_each do |ms|
-      raw_stats = fetch_raw_stats(ms, start_time, end_time, interval)
+      begin
+        raw_stats = fetch_raw_stats(ms, start_time, end_time, interval)
```

Comment:
I just know of [TemporaryFailure](https://github.com/durandom/manageiq/blob/a849dbec3d95d15129487cb5165f64a5146b4e18/app/models/miq_worker/runner.rb#L5-L5) which is used to throttle the workers.
But @jrafanie probably knows more :smile: